src/main/java/org/apache/sling/resourceresolver/impl/ResourceResolverFactoryActivator.java

```diff
@@ -306,27 +307,29 @@ protected void activate(final BundleContext bundleContext, final ResourceResolve
         }
 
         // optimize alias path allow list
-        this.aliasPathAllowList.clear();
         String[] aliasPathPrefix = config.resource_resolver_allowed_alias_locations();
         if ( aliasPathPrefix != null ) {
+            final Set<String> prefixSet = new TreeSet<>();
             for(final String prefix : aliasPathPrefix) {
                 String value = prefix.trim();
-                    if (!value.isEmpty()&&value.startsWith("/")) { // absolute path should be given
-                        if (value.endsWith("/")) {
-                            this.aliasPathAllowList.add(value);
-                        } else {
-                            this.aliasPathAllowList.add(value + "/");
-                        }
+                if (!value.isEmpty()) {
+                    if ( value.endsWith("/") ) {
+                        prefixSet.add(value);
+                    } else {
+                        prefixSet.add(value + "/");
                     }
+                }
+            }
+            if ( !prefixSet.isEmpty()) {
+                this.aliasPathAllowList = Collections.unmodifiableSet(prefixSet);
             }
-            Collections.sort(this.aliasPathAllowList);
         }
 
         // vanity path white list
         this.vanityPathWhiteList = null;
         String[] vanityPathPrefixes = config.resource_resolver_vanitypath_whitelist();
         if ( vanityPathPrefixes != null ) {
-            final List<String> prefixList = new ArrayList<>();
+             List<String> prefixList = new ArrayList<>();
```

Comment:
I like `final` variables. It tells me I don't have to scan the code for unexpected re-assignments. Also, this line is outside the scope of this PR if I understand correctly.

Reply:
yes, it is not in the scope of PR.
updated changes